app/state.go

```diff
@@ -1789,13 +1795,18 @@ func loadStateData(txn store.Transaction, key string) (datastore.StateData, erro
 		return sd, err
 	}
 
-	// we are relying on the fact that Unmarshal will decode all and only the fields
+	// We are relying on the fact that Unmarshal will decode all and only the fields
 	// that it can find in the destination type.
 	err = json.Unmarshal(data, &sd)
 	if err != nil {
 		return sd, err
 	}
 
+	if sd.UpdateInfo.Artifact.Provides.ArtifactName == "" {
+		sd.UpdateInfo.Artifact.Provides.
+			ArtifactName = sd.UpdateInfo.Artifact.ArtifactNameAlias
```

Comment:
We also need to store `ArtifactNameAlias` when we save the data back to disk in `StoreStateDataAndTransaction`, otherwise downgrading to an old client won't work.